flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1487,7 +1487,7 @@ class JobManager(
           }
         }
 
-        eg.fail(cause)
+        eg.cancel()
```

Comment:
The problem is that `postStop()` is not only called when the cluster is cleanly shut down (and you want job state removed) but in some cases on master failure, where we need to be sure that state is not removed from ZooKeeper.

The `Integer.MAX_VALUE` issue is less critical...